src/methods/method_overloading/hard/q4/MainTest.java

```diff
@@ -0,0 +1,156 @@
+                Arguments.of(32917, 4, 57.3, 1.321, 131668, 1886144.1, 16, 1.745041),
+                Arguments.of(-43, -23, 43, 763, 989, -1849.0, 529, 582169.0),
+                Arguments.of(4738, -32, -4372.432, 584.432391, -151616, -2.0716582816E7, 1024, 341561.21964997693)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("twoIntegersInputProvider")
+    void twoIntegerCorrect(int a, int b, int area) throws Throwable {
+        Object[][] arguments = {
+                {a, int.class},
+                {b, int.class}
+        };
+        MethodTest m = new MethodTest(RectangleOverload.class, "rectangleArea", arguments);
+        Object output = m.callMethod();
+        String errorMsg = "Your rectangleArea() method does not calculate the correct area for two integers.";
+        CustomAssertions._assertEquals(area, output, errorMsg);
+    }
+
+    @ParameterizedTest
+    @MethodSource("oneIntegerInputProvider")
+    void oneIntegerCorrect(int a, int area) throws Throwable {
+        Object[][] arguments = {
+                {a, int.class}
+        };
+        MethodTest m = new MethodTest(RectangleOverload.class, "rectangleArea", arguments);
+        Object output = m.callMethod();
+        String errorMsg = "Your rectangleArea() method does not calculate the correct area for one integer.";
+        CustomAssertions._assertEquals(area, output, errorMsg);
+    }
+
+    @ParameterizedTest
+    @MethodSource("integerDoubleInputProvider")
+    void twoDoubleCorrect(int a, double b, double area) throws Throwable {
+        Object[][] arguments = {
+                {a, int.class},
+                {b, double.class}
+        };
+        MethodTest m = new MethodTest(RectangleOverload.class, "rectangleArea", arguments);
+        Object output = m.callMethod();
+        String errorMsg = "Your rectangleArea() method does not calculate the correct area for an integer and a double.";
+        CustomAssertions._assertEquals(area, output, 0.0001, errorMsg);
+    }
+
+    @ParameterizedTest
+    @MethodSource("oneDoubleInputProvider")
+    void oneDoubleCorrect(double a, double area) throws Throwable {
+        Object[][] arguments = {
+                {a, double.class}
+        };
+        MethodTest m = new MethodTest(RectangleOverload.class, "rectangleArea", arguments);
+        Object output = m.callMethod();
+        String errorMsg = "Your rectangleArea() method does not calculate the correct area for one double.";
+        CustomAssertions._assertEquals(area, output, 0.0001, errorMsg);
+    }
+
+    @ParameterizedTest
+    @MethodSource("mainMethodInputProvider")
+    void printsCorrectMessage(int a, int b, double x, double y, int twoInts, double intDouble, int oneInt, double oneDouble) {
+        runWithInput(a + " " + b + " " + x + " " + y);
+
+        assertEquals(twoInts, Integer.parseInt(getItemByName("int_int")), "Your program does not print the correct message for the first area.");
```

Comment:
Should be "... correct output for the area of the first rectangle". Apply this to all the error messages.